### ISSUE TITLE
Error using the schedule clause with static option

### ISSUE THREAD

**Issue by imadki** [pyccel/pyccel#1098]
**Describe the bug**

**To Reproduce**
The SCHEDULE clause with static option does not work

```python
if __name__ == "__main__":
    import numpy as np
    from pyccel.stdlib.internal.openmp import omp_get_thread_num, omp_get_num_threads

    n = 4096
    a = np.zeros(n)

    #$omp parallel private(rank,nb_tasks,i_min,i_max)                                                                                                                                                      
    rank = omp_get_thread_num(); nb_tasks=omp_get_num_threads(); i_min=n; i_max=0
    #$omp for nowait schedule(static, n/nb_tasks)                                                                                                                                                          

    for i in range(n):
        a[i] = 92290. + i
        i_min = min(i_min, i)
        i_max = max(i_min, i)

    print("Rank", rank, "i_min", i_min, "i_max", i_max)
    #$omp end parallel 
```

**Here is the equivalent in Fortran**
```fortran
program parallel

  !$ use OMP_LIB                                                                                                                                                                                           
  implicit none
  integer, parameter :: n=300
  real, dimension(n) :: a
  integer:: i, i_min, i_max, rank, nb_tasks

  !$OMP PARALLEL PRIVATE(rank,nb_tasks,i_min,i_max)                                                                                                                                                        
  rank=OMP_GET_THREAD_NUM() ; nb_tasks= OMP_GET_NUM_THREADS() ; i_min=n ; i_max=0

  !$OMP DO SCHEDULE(STATIC,n/nb_tasks)                                                                                                                                                                     
  do i = 1, n
     a(i) = 92290. + real(i) ; i_min=min(i_min,i) ; i_max=max(i_max,i)
  end do
  !$OMP END DO NOWAIT                                                                                                                                                                                      

  print *,"Rank : ",rank,"; i_min :",i_min,"; i_max :",i_max
  !$OMP END PARALLEL                                                                                                                                                                                       

end program parallel

```

**Upon the compilation stage**
```bash
...
File "/home/kissami/.local/lib/python3.8/site-packages/arpeggio/__init__.py", line 1718, in _nm_raise
    raise self.nm
arpeggio.NoMatch: Expected INT at position (1, 35) => 'e(static, *n/nb_tasks'.

...
File "/home/kissami/.local/lib/python3.8/site-packages/textx/model.py", line 294, in _parse
    raise TextXSyntaxError(message=text(e),
textx.exceptions.TextXSyntaxError: None:1:35: error: Expected INT at position (1, 35) => 'e(static, *n/nb_tasks'.
```

**Comment by EmilyBourne** [comment]
@nhamidn Can you confirm that this is the same bug stopping [this test](https://github.com/pyccel/pyccel/blob/master/tests/epyccel/test_epyccel_openmp.py#L201-L205) from working?

**Comment by EmilyBourne** [comment]
> @nhamidn Can you confirm that this is the same bug stopping [this test](https://github.com/pyccel/pyccel/blob/master/tests/epyccel/test_epyccel_openmp.py#L201-L205) from working?

This is not entirely due to the bug mentioned above.
There are 2 problems here:
1. pyccel can't read arithmetic statements in openmp commands
2. the schedule command is read incorrectly and assumes the second argument must be a literal integer

**Comment by EmilyBourne** [comment]
After PR #1101 this problem will have the following work-around:
```python
if __name__ == "__main__":
    import numpy as np
    from pyccel.stdlib.internal.openmp import omp_get_thread_num, omp_get_num_threads

    n = 4096
    a = np.zeros(n)

    #$omp parallel private(rank,nb_tasks,i_min,i_max)
    rank = omp_get_thread_num(); nb_tasks=omp_get_num_threads(); i_min=n; i_max=0
    schedule_size = int(n/nb_tasks)
    #$omp for nowait schedule(static, schedule_size)

    for i in range(n):
        a[i] = 92290. + i
        i_min = min(i_min, i)
        i_max = max(i_min, i)

    print("Rank", rank, "i_min", i_min, "i_max", i_max)
    #$omp end parallel
```

**Comment by nhamidn** [comment]
> @nhamidn Can you confirm that this is the same bug stopping [this test](https://github.com/pyccel/pyccel/blob/master/tests/epyccel/test_epyccel_openmp.py#L201-L205) from working?

Yes I confirm, the arithmetic expressions inside the clauses are not managed yet

**Comment by nhamidn** [comment]
There is an issue for this problem #590

**Comment by nhamidn** [comment]
> > @nhamidn Can you confirm that this is the same bug stopping [this test](https://github.com/pyccel/pyccel/blob/master/tests/epyccel/test_epyccel_openmp.py#L201-L205) from working?
> 
> This is not entirely due to the bug mentioned above. There are 2 problems here:
> 
> 1. pyccel can't read arithmetic statements in openmp commands
> 2. the schedule command is read incorrectly and assumes the second argument must be a literal integer

Exactly, when pyccel can read expressions inside the clause, the openmp grammar should work fine

**Comment by EmilyBourne** [comment]
As the workaround is now working. I think we can consider the first part of this issue as fixed by #1101 and the second part as a duplicate of #590 , so I think we can close this issue
